diff --git a/index.ts b/index.ts
index 3859b630f2e253b89a9302b6cc7c283c15f17d55..45576300293e259544be419cb93739d71f5f1899 100644
--- a/index.ts
+++ b/index.ts
@@ -7,7 +7,7 @@
 /// <reference path="./i18n.d.ts" />
 /// <reference path="./virtual.d.ts" />
 
-import mdx from '@astrojs/mdx';
+// import mdx from '@astrojs/mdx';
 import type { AstroIntegration } from 'astro';
 import { spawn } from 'node:child_process';
 import { dirname, relative } from 'node:path';
@@ -96,9 +96,9 @@ export default function StarlightIntegration({
 				if (!allIntegrations.find(({ name }) => name === '@astrojs/sitemap')) {
 					integrations.push(starlightSitemap(starlightConfig));
 				}
-				if (!allIntegrations.find(({ name }) => name === '@astrojs/mdx')) {
-					integrations.push(mdx({ optimize: true }));
-				}
+				// if (!allIntegrations.find(({ name }) => name === '@astrojs/mdx')) {
+				// 	integrations.push(mdx({ optimize: true }));
+				// }
 
 				// Add Starlight directives restoration integration at the end of the list so that remark
 				// plugins injected by Starlight plugins through Astro integrations can handle text and
