diff --git a/dist/index.js b/dist/index.js
index f7fb4209ffd57c4d6f794b4485cf16aa58ad7422..b39e691ec7df4fc2d4d11595ad72835cf400c457 100644
--- a/dist/index.js
+++ b/dist/index.js
@@ -66,8 +66,7 @@ function mdx(mdxOptions = {}) {
                 async transform(_, id) {
                   if (!id.endsWith("mdx"))
                     return;
-                  const { fileId } = getFileInfo(id, config);
-                  const code = await fs.readFile(fileId, "utf-8");
+                  const code = '# Hello world!';
                   const { data: frontmatter, content: pageContent } = parseFrontmatter(code, id);
                   const compiled = await mdxCompile(new VFile({ value: pageContent, path: id }), {
                     ...mdxPluginOpts,