diff --git a/dist/index.js b/dist/index.js
index f7fb4209ffd57c4d6f794b4485cf16aa58ad7422..e5cf8fa5316f52803e36c1c8b1b29d4101bff799 100644
--- a/dist/index.js
+++ b/dist/index.js
@@ -11,6 +11,7 @@ import {
   rehypeApplyFrontmatterExport
 } from "./plugins.js";
 import { getFileInfo, handleExtendsNotSupported, parseFrontmatter } from "./utils.js";
+console.log('~~~~~ RUNNING PATCHED VERSION OF MDX INTEGRATION ~~~~~');
 const RAW_CONTENT_ERROR = "MDX does not support rawContent()! If you need to read the Markdown contents to calculate values (ex. reading time), we suggest injecting frontmatter via remark plugins. Learn more on our docs: https://docs.astro.build/en/guides/integrations-guide/mdx/#inject-frontmatter-via-remark-or-rehype-plugins";
 const COMPILED_CONTENT_ERROR = "MDX does not support compiledContent()! If you need to read the HTML contents to calculate values (ex. reading time), we suggest injecting frontmatter via rehype plugins. Learn more on our docs: https://docs.astro.build/en/guides/integrations-guide/mdx/#inject-frontmatter-via-remark-or-rehype-plugins";
 function mdx(mdxOptions = {}) {
@@ -82,7 +83,7 @@ function mdx(mdxOptions = {}) {
                   });
                   return {
                     code: escapeViteEnvReferences(String(compiled.value)),
-                    map: compiled.map
+                    // map: compiled.map
                   };
                 }
               },