diff --git a/schema.ts b/schema.ts
index 5f73dfdc598c04651b9051e927a758eaf07e1aa6..20894e6fb9812a4f5f201314882f4faf35461026 100644
--- a/schema.ts
+++ b/schema.ts
@@ -1,5 +1,5 @@
 import { z } from 'astro/zod';
-import type { SchemaContext } from 'astro:content';
+import type { SchemaContext, ImageFunction } from 'astro:content';
 import { HeadConfigSchema } from './schemas/head';
 import { PrevNextLinkConfigSchema } from './schemas/prevNextLink';
 import { TableOfContentsSchema } from './schemas/tableOfContents';
@@ -11,9 +11,15 @@ export { i18nSchema } from './schemas/i18n';
 type IconName = keyof typeof Icons;
 const iconNames = Object.keys(Icons) as [IconName, ...IconName[]];
 
-export function docsSchema() {
-	return ({ image }: SchemaContext) =>
-		z.object({
+type AnyZodObject =
+	| z.AnyZodObject
+	| z.ZodUnion<[z.AnyZodObject, ...z.AnyZodObject[]]>
+	| z.ZodDiscriminatedUnion<string, z.AnyZodObject[]>
+	| z.ZodIntersection<z.AnyZodObject, z.AnyZodObject>;
+
+export function docsSchema<S extends AnyZodObject>({ extend }: { extend?: S } = {}) {
+	return ({ image }: { image: ImageFunction }) => {
+		const schema = z.object({
 			/** The title of the current page. Required. */
 			title: z.string(),
 
@@ -156,4 +162,6 @@ export function docsSchema() {
 			/** Pagefind indexing for this page - set to false to disable. */
 			pagefind: z.boolean().default(true),
 		});
+    return schema.and(extend || z.object({}));
+  };
 }