diff --git a/schema.ts b/schema.ts
index cece5949488c65823101d548a398e4b6e0e1e6db..f672047401e4d965cec8d6809c917fafc364aeea 100644
--- a/schema.ts
+++ b/schema.ts
@@ -8,9 +8,15 @@ import { HeroSchema } from './schemas/hero';
 import { SidebarLinkItemHTMLAttributesSchema } from './schemas/sidebar';
 export { i18nSchema } from './schemas/i18n';
 
-export function docsSchema() {
-	return (context: SchemaContext) =>
-		z.object({
+type AnyZodObject =
+	| z.AnyZodObject
+	| z.ZodUnion<[z.AnyZodObject, ...z.AnyZodObject[]]>
+	| z.ZodDiscriminatedUnion<string, z.AnyZodObject[]>
+	| z.ZodIntersection<z.AnyZodObject, z.AnyZodObject>;
+
+export function docsSchema<S extends AnyZodObject>({ extend }: { extend?: S } = {}) {
+	return (context: SchemaContext) => {
+		const schema = z.object({
 			/** The title of the current page. Required. */
 			title: z.string(),
 
@@ -104,4 +110,6 @@ export function docsSchema() {
 			/** Pagefind indexing for this page - set to false to disable. */
 			pagefind: z.boolean().default(true),
 		});
+    return schema.and(extend || z.object({}));
+  };    
 }