diff --git a/dist/index.js b/dist/index.js
index f7fb4209ffd57c4d6f794b4485cf16aa58ad7422..8d2946b4eae8ef9fcbcd0751aa0283024a2b6355 100644
--- a/dist/index.js
+++ b/dist/index.js
@@ -66,7 +66,13 @@ function mdx(mdxOptions = {}) {
                 async transform(_, id) {
                   if (!id.endsWith("mdx"))
                     return;
-                  const { fileId } = getFileInfo(id, config);
+                  const { fileId, fileUrl } = getFileInfo(id, config);
+                  return {
+                    code: `export const Content = () => <p>Hello world</p>;
+                          export const url = ${JSON.stringify(fileUrl)};
+                          export const file = ${JSON.stringify(fileId)};
+                          export default Content;`
+                  };
                   const code = await fs.readFile(fileId, "utf-8");
                   const { data: frontmatter, content: pageContent } = parseFrontmatter(code, id);
                   const compiled = await mdxCompile(new VFile({ value: pageContent, path: id }), {
@@ -89,6 +95,7 @@ function mdx(mdxOptions = {}) {
               {
                 name: "@astrojs/mdx-postprocess",
                 transform(code, id) {
+                  return;
                   if (!id.endsWith(".mdx"))
                     return;
                   const [moduleImports, moduleExports] = parseESM(code);