diff --git a/dist/index.js b/dist/index.js
index f7fb4209ffd57c4d6f794b4485cf16aa58ad7422..896fdc32891cf551aebe71a23b2ba4dbc3828fbe 100644
--- a/dist/index.js
+++ b/dist/index.js
@@ -77,7 +77,7 @@ function mdx(mdxOptions = {}) {
                     ],
                     recmaPlugins: [
                       ...mdxPluginOpts.recmaPlugins ?? [],
-                      () => recmaInjectImportMetaEnvPlugin({ importMetaEnv })
+                      // () => recmaInjectImportMetaEnvPlugin({ importMetaEnv })
                     ]
                   });
                   return {