diff --git a/dist/index.js b/dist/index.js
index f7fb4209ffd57c4d6f794b4485cf16aa58ad7422..864d5c40e7aa48f11f8221fd60da2c9b3b893330 100644
--- a/dist/index.js
+++ b/dist/index.js
@@ -14,6 +14,7 @@ import { getFileInfo, handleExtendsNotSupported, parseFrontmatter } from "./util
 const RAW_CONTENT_ERROR = "MDX does not support rawContent()! If you need to read the Markdown contents to calculate values (ex. reading time), we suggest injecting frontmatter via remark plugins. Learn more on our docs: https://docs.astro.build/en/guides/integrations-guide/mdx/#inject-frontmatter-via-remark-or-rehype-plugins";
 const COMPILED_CONTENT_ERROR = "MDX does not support compiledContent()! If you need to read the HTML contents to calculate values (ex. reading time), we suggest injecting frontmatter via rehype plugins. Learn more on our docs: https://docs.astro.build/en/guides/integrations-guide/mdx/#inject-frontmatter-via-remark-or-rehype-plugins";
 function mdx(mdxOptions = {}) {
+  let buildNum = 0;
   return {
     name: "@astrojs/mdx",
     hooks: {
@@ -66,8 +67,14 @@ function mdx(mdxOptions = {}) {
                 async transform(_, id) {
                   if (!id.endsWith("mdx"))
                     return;
-                  const { fileId } = getFileInfo(id, config);
-                  const code = await fs.readFile(fileId, "utf-8");
+                  const code = `---
+                  title: Use a CMS with Astro ${buildNum += 1}
+                  description: How to use a CMS to add content to Astro
+                  layout: ~/layouts/MainLayout.astro
+                  i18nReady: true
+                  ---
+                  
+                  **Ready to connect a Headless CMS to your Astro project?** Follow one of our guides to integrate a CMS.`;
                   const { data: frontmatter, content: pageContent } = parseFrontmatter(code, id);
                   const compiled = await mdxCompile(new VFile({ value: pageContent, path: id }), {
                     ...mdxPluginOpts,