diff --git a/components/RightSidebarPanel.astro b/components/RightSidebarPanel.astro
index 7e5ca5218391531a024d4f46e9164e7165beac08..ac45dd4a9291f90181021a4792834b076f39a1ec 100644
--- a/components/RightSidebarPanel.astro
+++ b/components/RightSidebarPanel.astro
@@ -1,7 +1,13 @@
+---
+import StarlightBanner from '~/components/StarlightBanner.astro'
+---
+
 <div class="right-sidebar-panel hidden lg:block">
 	<div class="container">
 		<slot />
 	</div>
+
+  <StarlightBanner/>
 </div>
 
 <style>
diff --git a/components/SiteTitle.astro b/components/SiteTitle.astro
index 8b8982ab890ee003b5db94d2b0ccbf2566d2d2cc..4d7d922385854ff23407a297367981e1bbb05439 100644
--- a/components/SiteTitle.astro
+++ b/components/SiteTitle.astro
@@ -23,7 +23,7 @@ if (config.logo) {
 	if (err) throw new Error(err);
 }
 
-const href = pathWithBase(Astro.props.locale || '/');
+const href = pathWithBase((Astro.props.locale || '') + '/getting-started/');
 ---
 
 <a {href} class="site-title flex">
diff --git a/layout/TwoColumnContent.astro b/layout/TwoColumnContent.astro
index 8ba8da0fdef59b92b2f883ee5fed9300df86c19f..5cebd9aeae93c505d2635aad3175fbc5f61a4e18 100644
--- a/layout/TwoColumnContent.astro
+++ b/layout/TwoColumnContent.astro
@@ -36,7 +36,6 @@ interface Props {
 			top: 0;
 			border-inline-start: 1px solid var(--sl-color-gray-6);
 			padding-top: var(--sl-nav-height);
-			width: 100%;
 			height: 100vh;
 			overflow-y: auto;
 			scrollbar-width: none;
diff --git a/schema.ts b/schema.ts
index b45dcc3a5e90dc3026de35e4a04b6a9b36c26528..aafd1ba6c139d980310310b1e4c1672345c6b3c9 100644
--- a/schema.ts
+++ b/schema.ts
@@ -25,9 +25,15 @@ type ImageFunction = () => z.ZodObject<{
 	>;
 }>;
 
-export function docsSchema() {
-	return ({ image }: { image: ImageFunction }) =>
-		z.object({
+type AnyZodObject =
+	| z.AnyZodObject
+	| z.ZodUnion<[z.AnyZodObject, ...z.AnyZodObject[]]>
+	| z.ZodDiscriminatedUnion<string, z.AnyZodObject[]>
+	| z.ZodIntersection<z.AnyZodObject, z.AnyZodObject>;
+
+export function docsSchema<S extends AnyZodObject>({ extend }: { extend?: S } = {}) {
+	return ({ image }: { image: ImageFunction }) => {
+		const schema = z.object({
 			/** The title of the current page. Required. */
 			title: z.string(),
 
@@ -150,4 +156,6 @@ export function docsSchema() {
 				})
 				.default({}),
 		});
+    return schema.and(extend || z.object({}));
+  };
 }