diff --git a/dist/index.js b/dist/index.js
index f7fb4209ffd57c4d6f794b4485cf16aa58ad7422..ad8e2cea6414a05658024f6041701ce7db1b29fb 100644
--- a/dist/index.js
+++ b/dist/index.js
@@ -7,10 +7,10 @@ import { VFile } from "vfile";
 import {
   getRehypePlugins,
   getRemarkPlugins,
-  recmaInjectImportMetaEnvPlugin,
   rehypeApplyFrontmatterExport
 } from "./plugins.js";
 import { getFileInfo, handleExtendsNotSupported, parseFrontmatter } from "./utils.js";
+console.log('~~~~~ USING PATCHED MDX INTEGRATIONS ~~~~~');
 const RAW_CONTENT_ERROR = "MDX does not support rawContent()! If you need to read the Markdown contents to calculate values (ex. reading time), we suggest injecting frontmatter via remark plugins. Learn more on our docs: https://docs.astro.build/en/guides/integrations-guide/mdx/#inject-frontmatter-via-remark-or-rehype-plugins";
 const COMPILED_CONTENT_ERROR = "MDX does not support compiledContent()! If you need to read the HTML contents to calculate values (ex. reading time), we suggest injecting frontmatter via rehype plugins. Learn more on our docs: https://docs.astro.build/en/guides/integrations-guide/mdx/#inject-frontmatter-via-remark-or-rehype-plugins";
 function mdx(mdxOptions = {}) {
@@ -77,7 +77,7 @@ function mdx(mdxOptions = {}) {
                     ],
                     recmaPlugins: [
                       ...mdxPluginOpts.recmaPlugins ?? [],
-                      () => recmaInjectImportMetaEnvPlugin({ importMetaEnv })
+                      // () => recmaInjectImportMetaEnvPlugin({ importMetaEnv })
                     ]
                   });
                   return {