diff --git a/.vscode/settings.json b/.vscode/settings.json
new file mode 100644
index 0000000000000000000000000000000000000000..7c2feb7ef4731d430facfcafcbac59e3ea861849
--- /dev/null
+++ b/.vscode/settings.json
@@ -0,0 +1,3 @@
+{
+  "editor.formatOnSave": false
+}
diff --git a/dist/remark-shiki.js b/dist/remark-shiki.js
index e19774151e85b4e1670c86655ad35f6066d93685..a71f99908e124996676a3a7e66016bae3d9774c0 100644
--- a/dist/remark-shiki.js
+++ b/dist/remark-shiki.js
@@ -1,9 +1,7 @@
 import { getHighlighter } from "shiki";
 import { visit } from "unist-util-visit";
-const highlighterCacheAsync = /* @__PURE__ */ new Map();
 const remarkShiki = async ({ langs = [], theme = "github-dark", wrap = false }) => {
-  const cacheID = typeof theme === "string" ? theme : theme.name;
-  let highlighterAsync = highlighterCacheAsync.get(cacheID);
+  let highlighterAsync = null;
   if (!highlighterAsync) {
     highlighterAsync = getHighlighter({ theme }).then((hl) => {
       hl.setColorReplacements({
@@ -21,7 +19,6 @@ const remarkShiki = async ({ langs = [], theme = "github-dark", wrap = false })
       });
       return hl;
     });
-    highlighterCacheAsync.set(cacheID, highlighterAsync);
   }
   const highlighter = await highlighterAsync;
   for (const lang of langs) {
@@ -58,7 +55,7 @@ const remarkShiki = async ({ langs = [], theme = "github-dark", wrap = false })
         );
       }
       node.type = "html";
-      node.value = html;
+      // node.value = html;
       node.children = [];
     });
   };