diff --git a/dist/assets/build/generate.js b/dist/assets/build/generate.js
index 7291366be2a6b024819fb3c0fbd2247b5135f164..a9a4c164b7c08106ee7b310a4239fb6d8d71127c 100644
--- a/dist/assets/build/generate.js
+++ b/dist/assets/build/generate.js
@@ -52,7 +52,7 @@ function getFullImagePath(originalFilePath, env) {
   );
 }
 async function generateImagesForPath(originalFilePath, transformsAndPath, env, queue) {
-  const originalImageData = await loadImage(originalFilePath, env);
+  const originalImageData = await loadImage(transformsAndPath.originalSrcPath, env);
   for (const [_, transform] of transformsAndPath.transforms) {
     await queue.add(async () => generateImage(originalImageData, transform.finalPath, transform.transform)).catch((e) => {
       throw e;
@@ -182,7 +182,7 @@ async function loadImage(path, env) {
     };
   }
   return {
-    data: await fs.promises.readFile(getFullImagePath(path, env)),
+    data: await fs.promises.readFile(path),
     expires: 0
   };
 }
