diff --git a/dist/generateOpenGraphImage.js b/dist/generateOpenGraphImage.js
index 3e2de49da09ec59f7bd19b5f92154f0d22a747a4..2fc82927dfaaee20502975f32b0fe7f61f1100c6 100644
--- a/dist/generateOpenGraphImage.js
+++ b/dist/generateOpenGraphImage.js
@@ -129,5 +129,6 @@ export async function generateOpenGraphImage({ title, description = '', dir = 'l
     // Render canvas to a buffer.
     const image = surface.makeImageSnapshot();
     const imageBytes = image.encodeToBytes(CanvasKit.ImageFormat[format], quality) || new Uint8Array();
+    surface.dispose();
     return Buffer.from(imageBytes);
 }
diff --git a/dist/routing.js b/dist/routing.js
index 9653e6cee1429322099e43d67d604fad81801d44..ba6011ac484490573d73f7bf6feae375546002e0 100644
--- a/dist/routing.js
+++ b/dist/routing.js
@@ -24,6 +24,7 @@ function createOGImageEndpoint({ getSlug = pathToSlug, ...opts }) {
     };
 }
 export function OGImageRoute(opts) {
+    console.log('!!!!!!!!!! USING PATCHED ASTRO-OG-IMAGE !!!!!!!!!!')
     return {
         getStaticPaths: makeGetStaticPaths(opts),
         get: createOGImageEndpoint(opts),