diff --git a/dist/content/internal.js b/dist/content/internal.js
index 697a57cdce15610a422f9a42e861f2f4d7d99737..cb9f11759651c0516ee4bbc2334e1c137e5f3bac 100644
--- a/dist/content/internal.js
+++ b/dist/content/internal.js
@@ -28,12 +28,17 @@ function createCollectionToGlobResultMap({
   }
   return collectionToGlobResultMap;
 }
+const cacheByCollection = new Map();
 function createGetCollection({
   collectionToEntryMap,
   collectionToRenderEntryMap
 }) {
   return async function getCollection(collection, filter) {
     const lazyImports = Object.values(collectionToEntryMap[collection] ?? {});
+    // TODO: cache on prod builds only
+    if (cacheByCollection.has(collection)) {
+			return cacheByCollection.get(collection);
+		}
     const entries = Promise.all(
       lazyImports.map(async (lazyImport) => {
         const entry = await lazyImport();
@@ -53,6 +58,7 @@ function createGetCollection({
         };
       })
     );
+    cacheByCollection.set(collection, entries);
     if (typeof filter === "function") {
       return (await entries).filter(filter);
     } else {